docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/OrdinaryLeastSquaresAdvanced.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+using Microsoft.ML.SamplesUtils;
+
+namespace Samples.Dynamic.Trainers.Regression
+{
+    public static class OrdinaryLeastSquaresAdvanced
+    {
+        // This example requires installation of additional nuget package <a href="https://www.nuget.org/packages/Microsoft.ML.Mkl.Components/">Microsoft.ML.Mkl.Components</a>.
+        // In this examples we will use the housing price dataset. The goal is to predict median home value.
+        // For more details about this dataset, please see https://archive.ics.uci.edu/ml/machine-learning-databases/housing/
+        public static void Example()
+        {
+            // Downloading a regression dataset from github.com/dotnet/machinelearning
+            string dataFile = Microsoft.ML.SamplesUtils.DatasetUtils.DownloadHousingRegressionDataset();
```

Comment:
>Microsoft.ML.SamplesUtils.DatasetUtils.DownloadHousingRegressionDa [](start = 30, length = 66)

should this go?

Reply:
No. I don't want to remove all existing pipeline examples. They can be helpful when creating 3rd or 4th examples to an API. If we remove them, we will have to design all new pipelines in the future.

---
In reply to: [275457584](https://github.com/dotnet/machinelearning/pull/3319#discussion_r275457584) [](ancestors = 275457584)

Reply:
I don't think Senja was asking you about removing this file. I think she was asking you to remove reference to `Microsoft.ML.SamplesUtils.DatasetUtils.DownloadHousingRegressionDataset`
You were the person who hated this methods most and wanted to have in-memory examples.
Well this one is not in-memory for sure :)

---
In reply to: [275581072](https://github.com/dotnet/machinelearning/pull/3319#discussion_r275581072) [](ancestors = 275581072,275457584)

Reply:
This is not referenced in any API docs. What's its relation with in-memory thing?

---
In reply to: [275581828](https://github.com/dotnet/machinelearning/pull/3319#discussion_r275581828) [](ancestors = 275581828,275581072,275457584)

Reply:
You were devoted fanatic of use in-memory samples EVERYWHERE.
What happened?

---
In reply to: [275582931](https://github.com/dotnet/machinelearning/pull/3319#discussion_r275582931) [](ancestors = 275582931,275581828,275581072,275457584)

Reply:
This is a record for saving something for future use. Yes, EVERYWHERE for API documents. I guess I don't violate this rule, do I?

---
In reply to: [275583602](https://github.com/dotnet/machinelearning/pull/3319#discussion_r275583602) [](ancestors = 275583602,275582931,275581828,275581072,275457584)